app/controllers/comments_controller.rb

```diff
@@ -1,21 +1,25 @@
 class CommentsController < ApplicationController
-  before_action :authenticate_to_submit, only: :new
+  include HackernewsClone::VoteHelper
+  include HackernewsClone::CommentHelper
```

Comment:
I think this is where `app/controllers/concerns` can step in to help out. something like `app/controllers/concerns/voteable_controller.rb` and `app/controllers/concerns/commentable_controller.rb` defining `VoteableController` and `CommentableController` respectively would be more inline with the Rails way™ and move away from the `Helper` suffix which might be confused with files in the `app/helpers` dir.

Reply:
I made this change, then I thought about things more and I don't think the VoteHelper method belongs in a concerns dir under controllers. The method is really a query method that looks up whether a vote exists which belongs to the current user and the parent instance that a given vote button is attached to. I'm working on how to write the code so that the parent and current user get passed from the view to the vote_presenter.